ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py

```diff
@@ -134,6 +135,123 @@ def _generate_diag_args(draw):
     return dtype_x, offset, dtype_padding_value, align, num_rows, num_cols
 
 
+@st.composite
+def _generate_eigh_tridiagonal_args(draw):
+    dtype, alpha = draw(
+        helpers.dtype_and_values(
+            min_dim_size=2,
+            min_num_dims=1,
+            max_num_dims=1,
+            min_value=2.0,
+            max_value=5,
+            available_dtypes=helpers.get_dtypes("float"),
+        )
+    )
+    beta_shape = len(alpha[0]) - 1
+    dtype, beta = draw(
+        helpers.dtype_and_values(
+            available_dtypes=dtype,
+            shape=(beta_shape,),
+            min_value=2.0,
+            max_value=5,
+        )
+    )
+
+    select = draw(st.sampled_from(("a", "i", "v")))
+    if select == "a":
+        select_range = None
+    elif select == "i":
+        range_slice = draw(
+            st.slices(beta_shape).filter(
+                lambda x: x.start
+                and x.stop
+                and x.step
+                and x.start >= 0
+                and x.stop >= 0
+                and x.step >= 0
+                and x.start < x.stop
+            )
+        )
+
+        select_range = [range_slice.start, range_slice.stop]
+    else:
+        range_start = draw(st.floats(1e-5, 1e4))
+        range_end = draw(st.floats(range_start + 0.1, 1e5))
+        select_range = [range_start, range_end]
+
+    eigvals_only = draw(st.booleans())
+    tol = draw(st.floats(1e-5, 1e-3) | st.just(None))
+    return dtype, alpha, beta, eigvals_only, select, select_range, tol
+
+
+# eigh_tridiagonal
+@handle_test(
+    fn_tree="eigh_tridiagonal",
+    args_packet=_generate_eigh_tridiagonal_args(),
+    test_gradients=st.just(False),
+)
+def test_eigh_tridiagonal(
+    *,
+    args_packet,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+):
+    dtype, alpha, beta, eigvals_only, select, select_range, tol = args_packet
+    test_flags.with_out = False
+    results = helpers.test_function(
+        ground_truth_backend="numpy",
+        test_flags=test_flags,
+        fw=backend_fw,
+        fn_name=fn_name,
+        on_device=on_device,
+        rtol_=1e-2,
+        atol_=1e-2,
+        input_dtypes=dtype,
+        alpha=alpha[0],
+        beta=beta[0],
+        eigvals_only=eigvals_only,
+        select=select,
+        select_range=select_range,
+        tol=tol,
+        test_values=eigvals_only,
+        return_flat_np_arrays=True,
+    )
+    if results is None:
+        return
+    ret_np_flat, ret_from_np_flat = results
```

Comment:
The second one would come form the ground truth framework, you can abbreviate it at `gt`